rust-src/concordium_base/src/id/id_verifier.rs

```diff
@@ -362,13 +432,112 @@ mod tests {
         let upper = AttributeKind("20000103".to_string());
         let value = Value::<G1>::new(attribute.to_field_element());
         let (commitment, randomness) = keys.commit(&value, &mut csprng);
-        let maybe_proof =
-            prove_attribute_in_range(gens, &keys, &attribute, &lower, &upper, &randomness);
+        let mut transcript = RandomOracle::domain("Test");
+        let maybe_proof = prove_attribute_in_range(
+            ProofVersion::Version1,
+            &mut transcript.split(),
+            &mut thread_rng(),
+            gens,
+            &keys,
+            &attribute,
+            &lower,
+            &upper,
+            &randomness,
+        );
         if let Some(proof) = maybe_proof {
             assert_eq!(
-                verify_attribute_range(&keys, gens, &lower, &upper, &commitment, &proof),
+                verify_attribute_range(
+                    ProofVersion::Version1,
+                    &mut transcript,
+                    &keys,
+                    gens,
+                    &lower,
+                    &upper,
+                    &commitment,
+                    &proof
+                ),
                 Ok(()),
-                "Incorrect range proof."
+                "Incorrect version 1 range proof."
+            );
+        } else {
+            panic!("Failed to produce proof.");
```

Comment:
Should we also version here? I.e.,
```suggestion
            panic!("Failed to produce version 1 proof.");
```